sys/shell/commands/sc_cryptoauthlib.c

```diff
@@ -0,0 +1,238 @@
+        for (int j = 0; j < 4; j++) {
+            get_bin(binary, data[data_count]);
+            printf("%s ", binary);
+            data_count++;
+        }
+        printf("LKU%d LKU%d LKU%d LKU%d\n", i, i+1, i+2, i+3);
+    }
+
+    printf("%03d:%03d ", data_count, data_count+3);
+    for (int i = 0; i < 4; i++) {
+        get_bin(binary, data[data_count]);
+        printf("%s ", binary);
+        data_count++;
+    }
+    printf("UE SEL LV LC\n");
+
+    printf("%03d:%03d ", data_count, data_count+3);
+    for (int i = 0; i < 4; i++) {
+        get_bin(binary, data[data_count]);
+        printf("%s ", binary);
+        data_count++;
+    }
+    printf("SL0 SL1 RFU0 RFU1\n");
+
+    printf("%03d:%03d ", data_count, data_count+3);
+    for (int i = 0; i < 4; i++) {
+        get_bin(binary, data[data_count]);
+        printf("%s ", binary);
+        data_count++;
+    }
+    printf("X509-0 X509-1 X509-2 X509-3\n");
+
+    for (int i = 0; i < 32; i += 4) {
+        static int key_cnt = 0;
+        printf("%03d:%03d ", data_count, data_count+3);
+        for (int j = 0; j < 4; j++) {
+            get_bin(binary, data[data_count]);
+            printf("%s ", binary);
+            data_count++;
+        }
+        printf("KC%d KC%d ", key_cnt, key_cnt);
+        key_cnt++;
+        printf("KC%d KC%d\n", key_cnt, key_cnt);
+        key_cnt++;
+    }
+
+    return 0;
+}
+
+int _check_lock_config(void)
+{
+    bool is_locked = false;
+    atcab_is_locked(LOCK_ZONE_CONFIG, &is_locked);
+
+    if (is_locked) {
+        printf("Config zone is locked\n");
+    }
+    else {
+        printf("Config zone is not locked\n");
+    }
+
+    return 0;
+}
+
+int _check_lock_data(void)
+{
+    bool is_locked = false;
+    atcab_is_locked(LOCK_ZONE_DATA, &is_locked);
+
+    if (is_locked) {
+        printf("Data zone is locked\n");
+    }
+    else {
+        printf("Data zone is not locked\n");
+    }
+
+    return 0;
+}
+
+int _lock_config(void)
+{
+    bool is_locked = false;
+    atcab_is_locked(LOCK_ZONE_CONFIG, &is_locked);
+
+    if (is_locked) {
+        printf("Error: Config zone is already locked\n");
+        return 1;
+    }
+
+    if (atcab_lock_config_zone() != ATCA_SUCCESS) {
+        printf("Error: Locking failed\n");
+        return 1;
+    }
+
+    printf("Device successfully locked\n");
+    return 0;
+}
+
+int _cryptoauth(int argc, char **argv)
+{
+    if (argc > 1) {
+        if ((strcmp(argv[1], "read") == 0)) {
+            return _read_config();
+        }
+        else if ((strcmp(argv[1], "lock_c") == 0)) {
+            return _lock_config();
+        }
+        else if ((strcmp(argv[1], "check_lc") == 0)) {
+            return _check_lock_config();
+        }
+        else if ((strcmp(argv[1], "check_ld") == 0)) {
+            return _check_lock_data();
+        }
+    }
+    else {
+        printf("* read - read atca device's config zone\n");
+        printf("* lock_c - lock atca device's config zone\n");
```

Comment:
Locking the config zone is irreversible right? If so, I suggest to add a warning here